transformations/transformations/scc_cuf.py

```diff
@@ -0,0 +1,867 @@
+
+    Parameters
+    ----------
+    routine: :any:`Subroutine`
+        The subroutine (e.g. the driver) to increase the heap size
+    """
+    vtype = SymbolAttributes(types.BasicType.INTEGER, kind=sym.IntrinsicLiteral("cuda_count_kind"))
+    routine.spec.append(ir.VariableDeclaration((sym.Scalar("cudaHeapSize", type=vtype),)))
+
+    assignment_lhs = routine.variable_map["istat"]
+    assignment_rhs = sym.InlineCall(function=sym.ProcedureSymbol(name="cudaDeviceSetLimit", scope=routine),
+                                    parameters=(sym.IntrinsicLiteral("cudaLimitMallocHeapSize"),
+                                                routine.variable_map["cudaHeapSize"]))
+
+    routine.body.prepend(ir.Assignment(lhs=assignment_lhs, rhs=assignment_rhs))
+    routine.body.prepend(ir.Comment(''))
+
+    # TODO: heap size, to be calculated?
+    routine.body.prepend(
+        ir.Assignment(lhs=routine.variable_map["cudaHeapSize"], rhs=sym.Product((10, 1024, 1024, 1024))))
+
+
+def remove_pragmas(routine):
+    """
+    Remove all pragmas.
+
+    Parameters
+    ----------
+    routine: :any:`Subroutine`
+        The subroutine in which to remove all pragmas
+    """
+    pragma_map = {}
+    for pragma in FindNodes(ir.Pragma).visit(routine.body):
+        pragma_map[pragma] = None
+    routine.body = Transformer(pragma_map).visit(routine.body)
+
+
+def get_integer_variable(routine, name):
+    """
+    Find a local variable in the routine, or create an integer-typed one.
+
+    Parameters
+    ----------
+    routine : :any:`Subroutine`
+        The subroutine in which to find the variable
+    name : string
+        Name of the variable to find the in the routine.
+    """
+    if name in routine.variable_map:
+        v_index = routine.variable_map[name]
+    else:
+        dtype = SymbolAttributes(BasicType.INTEGER)
+        v_index = sym.Variable(name=name, type=dtype, scope=routine)
+    return v_index
+
+
+def kernel_remove_vector_loops(routine, horizontal):
+    """
+    Remove all vector loops over the specified dimension.
+
+    Parameters
+    ----------
+    routine: :any:`Subroutine`
+        The subroutine in the vector loops should be removed.
+    horizontal : :any:`Dimension`
+        The dimension specifying the horizontal vector dimension
+    """
+    loop_map = {}
+    for loop in FindNodes(ir.Loop).visit(routine.body):
+        if loop.variable == horizontal.index:
+            loop_map[loop] = loop.body
+    routine.body = Transformer(loop_map).visit(routine.body)
+
+
+# TODO: correct "definition" of elemental/pure routines ...
+def is_elemental(routine):
+    """
+    Check whether :any:`Subroutine` ``routine`` is an elemental routine.
+
+    Need for distinguishing elemental and non-elemental function to transform
+    those in a different way.
+
+    Parameters
+    ----------
+    routine: :any:`Subroutine`
+        The subroutine to check whether elemental
+    """
+    if "ELEMENTAL" in routine.prefix:
```

Comment:
I agree. 
However, I thought we may need to discuss this more generally (what elemental (and/or pure) subroutines/functions are possible for which Fortran standard...) ... Maybe this could be even a utility function?